src/accelerator.rs

```diff
@@ -0,0 +1,281 @@
+        mods
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct AcceleratorParseError(String);
+
+impl std::fmt::Display for AcceleratorParseError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "[AcceleratorParseError]: {}", self.0)
+    }
+}
+
+fn parse_accelerator(accelerator_string: &str) -> Result<Accelerator, AcceleratorParseError> {
+    let mut mods = Modifiers::empty();
+    let mut key = Code::Unidentified;
+
+    for raw in accelerator_string.split('+') {
+        let token = raw.trim().to_string();
+        if token.is_empty() {
+            return Err(AcceleratorParseError(
+                "Unexpected empty token while parsing accelerator".into(),
+            ));
+        }
+
+        if key != Code::Unidentified {
+            // at this point we already parsed the modifiers and found a main key but
+            // the function received more then one main key or it is not in the right order
+            // examples:
+            // 1. "Ctrl+Shift+C+A" => only one main key should be allowd.
+            // 2. "Ctrl+C+Shift" => wrong order
+            return Err(AcceleratorParseError(format!(
+                "Unexpected accelerator string format: \"{}\"",
+                accelerator_string
+            )));
+        }
+
+        match token.to_uppercase().as_str() {
+            "OPTION" | "ALT" => {
+                mods.set(Modifiers::ALT, true);
+            }
+            "CONTROL" | "CTRL" => {
+                mods.set(Modifiers::CONTROL, true);
+            }
+            "COMMAND" | "CMD" | "SUPER" => {
+                mods.set(Modifiers::SUPER, true);
+            }
+            "SHIFT" => {
+                mods.set(Modifiers::SHIFT, true);
+            }
+            "COMMANDORCONTROL" | "COMMANDORCTRL" | "CMDORCTRL" | "CMDORCONTROL" => {
+                #[cfg(target_os = "macos")]
+                mods.set(Modifiers::SUPER, true);
+                #[cfg(not(target_os = "macos"))]
+                mods.set(Modifiers::CONTROL, true);
+            }
+            _ => {
+                if let Ok(code) = Code::from_str(token.as_str()) {
+                    match code {
+                        Code::Unidentified => {
+                            return Err(AcceleratorParseError(format!(
+                                "Couldn't identify \"{}\" as a valid `Code`",
+                                token
+                            )))
+                        }
+                        _ => key = code,
+                    }
+                } else {
+                    return Err(AcceleratorParseError(format!(
+                        "Couldn't identify \"{}\" as a valid `Code`",
+                        token
+                    )));
+                }
+            }
+        }
+    }
+
+    Ok(Accelerator { key, mods })
+}
+
+#[test]
+fn test_parse_accelerator() {
+    assert_eq!(
+        parse_accelerator("CTRL+KeyX").unwrap(),
```

Comment:
This notation of `KeyX` will definitely break tauri so temporarily until tauri v2, we are gonna need to make a compatibility layer but not sure whether it should be in muda or tauri.
cc @lucasfernog

Reply:
How about we just fork the types in `keyboard_types` then. I feel like it doesn't need to stick to origin crate so hard.
We only need its `Code` and `Modifiers` types.

Reply:
No need to fork, we just need to have an internal function that takes the `Code` and converts it to a string that is compatible with current tao implementation. We only need to keep it around until tauri v2